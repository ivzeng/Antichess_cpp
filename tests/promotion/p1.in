h2d1

a7c2

b2b4

c2b1q
a1b1
undo
undo
c2b1b
a1b1
undo
undo
c2b1n
a1b1
undo
undo
c2b1r
a1b1
undo
undo