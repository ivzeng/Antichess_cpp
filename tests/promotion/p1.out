round0:
8 R N B Q K B N R 
7 P P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - - - - - - - - 
3 - - - - - - - - 
2 p p p p p p p p 
1 r n b q k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: 
placeful moves: b1c3 b1a3 g1h3 g1f3 a2a4 a2a3 b2b4 b2b3 c2c4 c2c3 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 h2h4 h2h3 
Capture::act() is called
round0:
8 R N B Q K B N R 
7 P P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - - - - - - - - 
3 - - - - - - - - 
2 p p p p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: 
placeful moves: b8c6 b8a6 g8h6 g8f6 a7a5 a7a6 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
Capture::act() is called
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - - - - - - - - 
3 - - - - - - - - 
2 p p P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: h1h7 d1c2 
placeful moves: b1c3 b1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b2b4 b2b3 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
Basic::act() is called
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 c2d1q c2d1r c2d1b c2d1n c2b1q c2b1r c2b1b c2b1n 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r Q b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
Capture::act() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 - r b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
Capture::reverse() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r Q b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 c2d1q c2d1r c2d1b c2d1n c2b1q c2b1r c2b1b c2b1n 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r B b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
Capture::act() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 - r b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
Capture::reverse() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r B b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 c2d1q c2d1r c2d1b c2d1n c2b1q c2b1r c2b1b c2b1n 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r N b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
Capture::act() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 - r b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
Capture::reverse() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r N b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 c2d1q c2d1r c2d1b c2d1n c2b1q c2b1r c2b1b c2b1n 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r R b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
Capture::act() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 - r b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
Capture::reverse() is called
round2:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - - p p p p - 
1 r R b p k b n r 
  a b c d e f g h 
player(1)'s turn:
capture moves: a1b1 h1h7 
placeful moves: c1b2 c1a3 g1h3 g1f3 h1h2 h1h3 h1h4 h1h5 h1h6 a2a4 a2a3 b4b5 d2d4 d2d3 e2e4 e2e3 f2f4 f2f3 g2g4 g2g3 
round1:
8 R N B Q K B N R 
7 - P P P P P P P 
6 - - - - - - - - 
5 - - - - - - - - 
4 - p - - - - - - 
3 - - - - - - - - 
2 p - P p p p p - 
1 r n b p k b n r 
  a b c d e f g h 
player(2)'s turn:
capture moves: a8a2 c2d1q c2d1r c2d1b c2d1n c2b1q c2b1r c2b1b c2b1n 
placeful moves: b8c6 b8a6 g8h6 g8f6 a8a7 a8a6 a8a5 a8a4 a8a3 b7b5 b7b6 c7c5 c7c6 d7d5 d7d6 e7e5 e7e6 f7f5 f7f6 g7g5 g7g6 h7h5 h7h6 
